File: item-api/src/main/java/kamon/demo/tracing/item/model/Seller.java
package kamon.demo.tracing.item.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import lombok.Value;

@Value(staticConstructor = "of")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Seller {

    private Long id;

    private String firstName;
    private String lastName;
    private LocalDate birthday;
    private SellerCategory category;
    private LocalDateTime registrationDate;
    private BigDecimal grade;

}

File: seller-api/src/main/kotlin/kamon/demo/tracing/seller/model/Seller.kt
package kamon.demo.tracing.seller.model


import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime

data class Seller(val id: Long, val firstName: String, val lastName: String,
                  val birthday: LocalDate, val category: SellerCategory,
                  val registrationDate: LocalDateTime, val grade: BigDecimal,
                  val active: Boolean)

enum class SellerCategory {
    BRONZE, SILVER, GOLD, PLATINUM
}